### PR TITLE
test: fix state fixtures race condition

### PR DESCRIPTION
**Pull request by seaona**
## **Description**
There is a race condition in our testing setup, which causes that the expected fixtures state, not being there when we start the test. This has been surfaced in [this branch](https://github.com/MetaMask/metamask-extension/pull/28277), where the account tracker multi polling is being added.
The problem is that if we don't have the AccountTracker in state when  the `resetState` function is called (at the beginning of wallet loading) the balance will remain loading until we refresh the wallet.

Edit: performing the load state early in the test setup fixes the issue.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28421?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3627

## **Manual testing steps**

1. Check all tests continue to pass
2. Check this changes fix this branch ENS test https://github.com/MetaMask/metamask-extension/pull/28402/files#diff-1acb7898d60977530c97169551d22dbe477a4e3aeb74f1f14bf2eea0b4d75d35  . Alternatively, see videos below with before and after behaviours

### **Before**

https://github.com/user-attachments/assets/8f50ec04-cf96-478e-9c3c-dce54254a628

### **After**

https://github.com/user-attachments/assets/0f109b1a-9289-48d9-b337-d51890c9d448

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/helpers.js [hunk 1]
```diff
@@ -119,6 +119,9 @@ async function withFixtures(options, testSuite) {
       contractRegistry = ganacheSeeder.getContractRegistry();
     }
 
+    await fixtureServer.start();
+    await fixtureServer.loadJsonState(fixtures, contractRegistry);
+
```
Comment (seaona) [inline test/e2e/helpers.js:124]: placing the calls here is also needed in order to mitigate the balance loading issue. It seems we need the state also earlier

**Review by chloeYue — DISMISSED** [review]
LGTM

**Review by sahar-fehri — DISMISSED** [review]

File: test/e2e/fixture-server.js [hunk 2]
```diff
@@ -104,8 +104,11 @@ class FixtureServer {
   }
 
   loadJsonState(rawState, contractRegistry) {
-    const state = performStateSubstitutions(rawState, contractRegistry);
-    this._stateMap.set(CURRENT_STATE_KEY, state);
+    return new Promise((resolve) => {
```
Comment (Gudahtt) [inline test/e2e/fixture-server.js:107]: Are these changes needed? It looks like nothing async is happening here. Wrapping in a Promise would have no impact (except for making more confusing stack traces, if we get an error thrown in here)

Comment (Gudahtt) [reply test/e2e/fixture-server.js:107]: Also we should be avoiding Promise constructors _in general_, favoring async/await syntax. We only need Promise constructors for wrapping callback APIs, which isn't happening here.

But either way, I would expect this to not have any functional impact.

**Review by Gudahtt — CHANGES_REQUESTED** [review]
More clarify is needed on how this issue was resolved. If wrapping a synchronous function in a Promise constructor is enough to fix the issue, it suggests that there is another Promise somewhere that we're not awaiting when we should (as the only impact this would have is in introducing one more place where we're waiting for the Promise queue), which suggests that we might still have a race condition here.

If it was fixed just by moving where the fixture server was started and initialized, we can reduce this PR just to those changes

File: test/e2e/fixture-server.js (hunk 2 above)
Comment (seaona) [reply test/e2e/fixture-server.js]: whoa Mark, thank you! That was indeed not needed

https://github.com/user-attachments/assets/c447ec7d-2a63-4519-ad4c-26a437328a31

Comment (seaona) [reply test/e2e/fixture-server.js]: Noted :pencil2: , thank you for your great insights as always, clarifying and full of wisdom :pray:

**Comment by seaona** [comment]
:100: to your last comment @Gudahtt 


> If it was fixed just by moving where the fixture server was started and initialized, we can reduce this PR just to those changes

This was indeed the fix, the rest didn't have any effect. I verified that it works, and also saw that there was nothing async happening inside that function call neither

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by hjetpoluru — APPROVED** [review]
LGTM
